plugins/hls-class-plugin/src/Ide/Plugin/Class/Types.hs

```diff
@@ -112,15 +112,15 @@ instance NFData InstanceBindLensResult where
 type instance RuleResult GetInstanceBindLens = InstanceBindLensResult
 
 data Log
-  = LogImplementedMethods Class [T.Text]
+  = LogImplementedMethods Class ClassMinimalDef
   | LogShake Shake.Log
 
 instance Pretty Log where
   pretty = \case
     LogImplementedMethods cls methods ->
-      pretty ("Detected implemented methods for class" :: String)
+      pretty ("The following methods are missing" :: String)
         <+> pretty (show (getOccString cls) <> ":") -- 'show' is used here to add quotes around the class name
-        <+> pretty methods
+        <+> pretty (showSDocUnsafe $ ppr methods)
```

Comment:
I think that's fine for logging, `unsafe` doesn't mean it might crash, but rather we use default `DynFlags` which may be missing some information, such as import paths, packages in scope, etc... So, the displayed names may be inaccurate. However, I can double-check what we do in other locations

Reply:
Maybe we should get the more exact `DynFlags` from rule like `GetModSummary`, but, I agree, for logging, it should be fine as long as it won't crash.